### PR TITLE
add channel and playlist stats to server stats endpoint

### PR DESCRIPTION
**Pull request by rigelk**
## Description

Adds the number of local playlists, local channels and active local channels on various timeframes (akin to active users stats): daily/weekly/monthly.

## Has this been tested?

- [x] 👍 yes, I added tests to the test suite, though they might not be enough

### CODE REVIEW

File: server/models/video/video-channel.ts [hunk 1]
```diff
@@ -316,6 +316,53 @@ export class VideoChannelModel extends Model {
     return VideoChannelModel.count(query)
   }
 
+  static async getStats () {
+    function getActiveVideoChannels (days: number) {
+      const query = {
```
Comment (Chocobozzz) [inline server/models/video/video-channel.ts:321]: Please make a raw sql request instead

File: server/models/video/video-playlist.ts [hunk 2]
```diff
@@ -494,6 +500,21 @@ export class VideoPlaylistModel extends Model {
     return '/video-playlists/embed/' + this.uuid
   }
 
+  static async getStats () {
+    const totalLocalPlaylists = await VideoPlaylistModel.scope({
```
Comment (Chocobozzz) [inline server/models/video/video-playlist.ts:504]: Don't use this scope, just make a regular request on public local playlists

File: server/tests/api/server/stats.ts [hunk 3]
```diff
@@ -121,6 +124,10 @@ describe('Test stats (excluding redundancy)', function () {
       expect(data.totalDailyActiveUsers).to.equal(1)
       expect(data.totalWeeklyActiveUsers).to.equal(1)
       expect(data.totalMonthlyActiveUsers).to.equal(1)
+
+      expect(data.totalLocalDailyActiveVideoChannels).to.equal(1)
```
Comment (Chocobozzz) [inline server/tests/api/server/stats.ts:128]: Only count that active channels === 1 is not enough: create another channel and check it's taken into account after an upload and not before

Also please correctly test `totalLocalPlaylists`

**Review by Chocobozzz — CHANGES_REQUESTED** [review]

**Comment by rigelk** [comment]
@Chocobozzz still missing the `totalLocalPlaylists` tests but otherwise ready for re-review

File: server/models/video/video-channel.ts [hunk 4]
```diff
@@ -316,6 +316,47 @@ export class VideoChannelModel extends Model {
...
+        type: QueryTypes.SELECT as QueryTypes.SELECT,
+        raw: true
+      }
+
+      const query = `
+SELECT          Count(DISTINCT("VideoChannelModel"."id")) AS "count"
+FROM            "videoChannel"                            AS "VideoChannelModel"
+INNER JOIN      "video"                                   AS "Videos"
+ON              "VideoChannelModel"."id" = "Videos"."channelId"
+AND             ("Videos"."createdAt" > Now() - interval '${days}d')
```
Comment (Chocobozzz) [inline server/models/video/video-channel.ts:332]: `publishedAt`

File: server/tests/api/server/stats.ts [hunk 5]
```diff
@@ -135,6 +142,68 @@ describe('Test stats (excluding redundancy)', function () {
...
+      const res = await getStats(server.url)
+      const data: ServerStats = res.body
+      expect(data.totalLocalDailyActiveVideoChannels).to.equal(1)
+      expect(data.totalLocalWeeklyActiveVideoChannels).to.equal(1)
+      expect(data.totalLocalMonthlyActiveVideoChannels).to.equal(1)
+    }
+
+    {
+      const resChannel = await getMyUserInformation(server.url, server.accessToken)
+      const videoChannel = server.videoChannel = resChannel.body.videoChannels.find(v => v.name === videoChannelProperties.name)
```
Comment (Chocobozzz) [inline server/tests/api/server/stats.ts:169]: Please don't override `server.videoChannel`. Just use a global variable instead. Also, you can use the answer of `addVideoChannel` to get the channel id: https://github.com/Chocobozzz/PeerTube/blob/develop/server/tests/api/search/search-activitypub-videos.ts#L104

This should resolve the test api-2 failure

**Review by Chocobozzz — DISMISSED** [review]

**Comment by Chocobozzz** [comment]
Please fix tests

**Pull request merged**